src/core/thread/mle_router.cpp

```diff
@@ -1506,26 +1503,6 @@ void MleRouter::HandleTimeTick(void)
             routerStateUpdate = true;
         }
     }
-#if OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
-    // Delay register only when `mRouterSelectionJitterTimeout` is 0,
-    // that is, when the device has decided to stay as REED or Router.
-    else if (mBackboneRouterRegistrationDelay > 0)
```

Comment:
Adding a note on this change:
We replace this `else` check corresponding to the `if (mRouterSelectionJitterTimeout > 0)` with a `IsExpectedToBecomeRouterSoon()` check in the `bbr_local.cpp`.